applications/audio/broadcast_configuration_tool/src/main.c

```diff
@@ -0,0 +1,504 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/zbus/zbus.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/shell/shell.h>
+#include <nrfx_clock.h>
+
+#include "broadcast_source.h"
+#include "zbus_common.h"
+#include "macros_common.h"
+#include "bt_mgmt.h"
+#include "sd_card.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(main, CONFIG_MAIN_LOG_LEVEL);
+
+ZBUS_CHAN_DECLARE(bt_mgmt_chan);
+ZBUS_CHAN_DECLARE(sdu_ref_chan);
+
+struct bt_le_ext_adv *ext_adv;
+
+/* Buffer for the UUIDs. */
+#define EXT_ADV_UUID_BUF_SIZE (128)
+NET_BUF_SIMPLE_DEFINE_STATIC(uuid_data, EXT_ADV_UUID_BUF_SIZE);
+NET_BUF_SIMPLE_DEFINE_STATIC(uuid_data2, EXT_ADV_UUID_BUF_SIZE);
+
+/* Buffer for periodic advertising BASE data. */
+NET_BUF_SIMPLE_DEFINE_STATIC(base_data, 128);
+NET_BUF_SIMPLE_DEFINE_STATIC(base_data2, 128);
+
+/* Extended advertising buffer. */
+static struct bt_data ext_adv_buf[CONFIG_BT_ISO_MAX_BIG][CONFIG_EXT_ADV_BUF_MAX];
+
+/* Periodic advertising buffer. */
+static struct bt_data per_adv_buf[CONFIG_BT_ISO_MAX_BIG];
+
+#if (CONFIG_AURACAST)
+/* Total size of the PBA buffer includes the 16-bit UUID, 8-bit features and the
+ * meta data size.
+ */
+#define BROADCAST_SRC_PBA_BUF_SIZE                                                                 \
+	(BROADCAST_SOURCE_PBA_HEADER_SIZE + CONFIG_BT_AUDIO_BROADCAST_PBA_METADATA_SIZE)
+
+#define PRESET_NAME_MAX 8
+
+/* Number of metadata items that can be assigned. */
+#define BROADCAST_SOURCE_PBA_METADATA_VACANT                                                       \
+	(CONFIG_BT_AUDIO_BROADCAST_PBA_METADATA_SIZE / (sizeof(struct bt_data)))
+
+/* Make sure pba_buf is large enough for a 16bit UUID and meta data
+ * (any addition to pba_buf requires an increase of this value)
+ */
+uint8_t pba_data[CONFIG_BT_ISO_MAX_BIG][BROADCAST_SRC_PBA_BUF_SIZE];
+
+static struct bt_bap_lc3_preset lc3_preset_8_2_1 = BT_BAP_LC3_BROADCAST_PRESET_8_2_1(
+	BT_AUDIO_LOCATION_FRONT_LEFT | BT_AUDIO_LOCATION_FRONT_RIGHT, BT_AUDIO_CONTEXT_TYPE_MEDIA);
```

Comment:
Just curious, would there be a case that only mono channel output?

Reply:
Yes, the location is not actually written here. This will be overwritten at a later stage, but the preset requires location to be set. I guess we could set it to 0 here and add a comment